File: game_client/cs205FINALPROJECT/src/Player.java
class Player
{
	private String playername;
	private boolean isHuman;
	private int id;
	private int score;
	public Hand myHand;
	
	//constructor
	public Player(boolean isHuman, int id, String name, Hand myHand)
	{
		this.isHuman = isHuman;
		this.id = id;
		this.playername = name;
		this.myHand = myHand;
	}
	
	public void setHand(Hand myHand)
	{
		this.myHand = myHand;
	}
	
	public Hand getHand()
	{
		return this.myHand;
	}	
	
	public void setHuman(boolean isHuman)
	{
		this.isHuman = isHuman;
	}
	public void setScore(int score)
	{
		this.score = score;
	}
	public void setID(int id)
	{
		this.id = id;
	}
	public void setName(String name)
	{
		playername = name;
	}
	public int getScore()
	{
		return score;
	}
	public boolean getHumanity()
	{
		return isHuman;
	}
	public int getID()
	{
		return id;
	}
	public String getName()
	{
		return playername;
	}
}

File: game_server/Models/Player.php
<?php
class Player{
	public $playerName;
	public $isHuman; 
	public $id;
	public $score;
	public $hand;
	public $totalScore;
	public $roundsWon;

	function __construct($jsonPlayer){
		$handJson = $jsonPlayer->myHand->hand;
		$this->playerName = $jsonPlayer->playername;
		$this->id = $this->emailToUserID($this->playerName);
		$this->isHuman = $jsonPlayer->isHuman;
		$this->score = $jsonPlayer->score;
		$this->totalScore = $jsonPlayer->totalScore;
		$numCards = count($handJson);
		for($ii=0; $ii<$numCards; $ii++){
			$this->hand[$ii] = array(
				'rank'=>$handJson[$ii]->rank,
				'faceup'=>$handJson[$ii]->faceup,
				'isSpecial'=>$handJson[$ii]->isSpecial
			);
		}
	}// end construct

	public function getScore(){
		return $this->score;
	}

	public function getID(){
		return $this->id;
	}

	public function emailToUserID(){
		$array = array(
			'tableName'=>'tblUserAccount',
			'fldEmail'=>$this->playerName
		);
		$dbWrapper=new InteractDB('select', $array);
		if(count($dbWrapper->returnedRows) > 0){
			$userID = $dbWrapper->returnedRows[0]['pkUserID'];
			// $userID = 888;
		}else{
			$userID = 666;
		}

		return $userID;
	} // end emailToUserID()
} // end Player class def
?>